evalml/automl/engines/sequential_engine.py

```diff
@@ -0,0 +1,68 @@
+from evalml.automl.engines import EngineBase
+
+
+class SequentialEngine(EngineBase):
+    """The default engine for the AutoML search. Evaluates pipelines locally and pipeline batches sequentially."""
+    def __init__(self):
+        super().__init__()
+        self.name = "Sequential Engine"
+
+    def evaluate_batch(self, pipeline_batch, result_callback=None, log_pipeline=False):
+        """ Evaluates a batch of pipelines in order.
+
+        Arguments:
+            pipeline_batch (list(PipelineBase)): A batch of pipelines to be fitted and evaluated.
+            result_callback (callable): Function called once the pipeline is finished evaluating to store the results. If None, results will only be returned.
+            log_pipeline (bool): If True, log the pipeline and relevant information before evaluation.
+
+        Returns:
+            list(PipelineBase), list(dict), list(PipelineBase): A list of evaluated pipelines, the results of the evaluated pipelines, and the remaining pipelines in the batch that were not evaluated.
+        """
+        super().evaluate_batch()
+        completed_pipelines = []
+        evaluation_results = []
+        while len(pipeline_batch) > 0:
+            pipeline = pipeline_batch.pop()
+            try:
+                if not self.automl._check_stopping_condition(self.automl._start):
+                    return completed_pipelines, evaluation_results, []
+                if log_pipeline:
+                    self.log_pipeline(pipeline)
+                if self.automl.start_iteration_callback:
+                    self.automl.start_iteration_callback(pipeline.__class__, pipeline.parameters, self)
+                pipeline, evaluation_result = self._compute_cv_scores(pipeline, self.automl, self.X, self.y)
+                if result_callback:
+                    self._add_result_callback(result_callback, pipeline, evaluation_result)
```

Comment:
I originally had it so that `_add_result` is called after the batch of pipelines is returned (which I think is the design that you mentioned). However, this design would mean that we could only update the search iteration plot after every batch rather than after every pipeline like we have in main. We could update the way that the pipeline iteration plot works so that it can update at a per-pipeline level by taking in a result. However, this would become messy quickly since it would involve having to combine results from the current search results and the finished results of the current batch. I figured it would be easier to have a function in `EngineBase` that has the callback + updates the plot rather than try to change how plotting works. 

I agree with the renaming since the current implementation is updating the plot + it makes it clear where the callback is going to be called. Updated in newest commit.